calico/felix/felix.py

```diff
@@ -93,40 +93,54 @@ def _main_greenlet(config):
 
-        v6_raw_updater = IptablesUpdater("raw", ip_version=6, config=config)
-        v6_filter_updater = IptablesUpdater("filter", ip_version=6,
-                                            config=config)
-        v6_nat_updater = IptablesUpdater("nat", ip_version=6, config=config)
-        v6_ipset_mgr = IpsetManager(IPV6, config)
-        v6_rules_manager = RulesManager(config,
-                                        6,
-                                        v6_filter_updater,
-                                        v6_ipset_mgr)
-        v6_dispatch_chains = DispatchChains(config, 6, v6_filter_updater)
-        v6_fip_manager = FloatingIPManager(config, 6, v6_nat_updater)
-        v6_ep_manager = EndpointManager(config,
-                                        IPV6,
-                                        v6_filter_updater,
-                                        v6_dispatch_chains,
-                                        v6_rules_manager,
-                                        v6_fip_manager,
-                                        etcd_api.status_reporter)
-        cleanup_mgr = CleanupManager(
-            config,
-            [v4_filter_updater, v6_filter_updater, v4_nat_updater],
-            [v4_ipset_mgr, v6_ipset_mgr]
-        )
-        update_splitter = UpdateSplitter(
-            [
-                v4_ipset_mgr, v6_ipset_mgr,
-                v4_rules_manager, v6_rules_manager,
-                v4_ep_manager, v6_ep_manager,
-                v6_raw_updater,
-                v4_masq_manager,
-                v4_nat_updater, v6_nat_updater,
-                cleanup_mgr,
-            ]
-        )
+        v6_enabled = os.path.exists("/proc/sys/net/ipv6")
+        if v6_enabled:
+            v6_raw_updater = IptablesUpdater("raw", ip_version=6, config=config)
+            v6_filter_updater = IptablesUpdater("filter", ip_version=6,
+                                                config=config)
+            v6_nat_updater = IptablesUpdater("nat", ip_version=6, config=config)
+            v6_ipset_mgr = IpsetManager(IPV6, config)
+            v6_rules_manager = RulesManager(config,
+                                            6,
+                                            v6_filter_updater,
+                                            v6_ipset_mgr)
+            v6_dispatch_chains = DispatchChains(config, 6, v6_filter_updater)
+            v6_fip_manager = FloatingIPManager(config, 6, v6_nat_updater)
+            v6_ep_manager = EndpointManager(config,
+                                            IPV6,
+                                            v6_filter_updater,
+                                            v6_dispatch_chains,
+                                            v6_rules_manager,
+                                            v6_fip_manager,
+                                            etcd_api.status_reporter)
+            cleanup_mgr = CleanupManager(
```

Comment:
There's a lot of duplication between this and the branch below.  Please can you refactor so that there's only one call to each constructor, taking the list of actors and then add the v6 actors to the list inside the conditional.